xds/internal/resolver/xds_resolver_test.go

```diff
@@ -857,12 +859,94 @@ func (s) TestXDSResolverWRR(t *testing.T) {
 		picks[clustermanager.GetPickedClusterForTesting(res.Context)]++
 		res.OnCommitted()
 	}
-	want := map[string]int{"A": 10, "B": 20}
+	want := map[string]int{"cluster:A": 10, "cluster:B": 20}
 	if !reflect.DeepEqual(picks, want) {
 		t.Errorf("picked clusters = %v; want %v", picks, want)
 	}
 }
 
+func init() {
```

Comment:
Can you move this & related tests to a new file?  This one is already too big.

Maybe `cluster_specifier_plugin_test.go`?